.github/workflows/tests.yml

```diff
@@ -0,0 +1,72 @@
+name: Unit tests
+
+on: [push]
+
+env:
+  RUST_BACKTRACE: full
+
+jobs:
+  unit:
+    name: "Unit Tests"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@main
+      - name: Cache Rust build
+        uses: actions/cache@v2
+        env:
+          cache-name: rust-build
+        with:
+          path: ~/.cargo
+          key: ${{ runner.os }}-build-${{ hashFiles('Cargo.toml') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ hashFiles('Cargo.toml') }}
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
+      - name: Install Rust compiler
+        uses: actions-rs/toolchain@v1
+        with:
+            toolchain: stable
+            default: true
+            override: true
+            components: rustfmt, clippy
+
+      - name: "Build"
+        run: cargo build --release
+
+      - name: "Unit tests"
+        run: cargo test --release
+
+      - name: "Synthesize rational rules"
+        run: |
+          cargo rational --iters 2 --variables 3 --num-fuzz 50 --do-final-run --outfile rat.json
```

Comment:
i think it's worth saving the result jsons in a designated directory.

Reply:
I know there is `tests/` but that's the expected ones right? What if we make a `tests/expected` and `tests/actual`? Thoughts?

Reply:
or maybe we can call it `out/` for the new ones and `tests/` for the pregenerated ones..

Reply:
Makes sense to me. I'll probably go with `out/` and `tests/`.